frontend/src/services/authentication_handler_service/authentication_handler_service.ts

```diff
@@ -50,20 +82,38 @@ export class AuthenticationHandlerService {
    * @return false if the user is already signed out, true if successful
    */
   async signOut(): Promise<boolean> {
-    const authInstance = window.gapi.auth2.getAuthInstance();
-    if (!authInstance) {
+    const auth = this.getAuthInstance();
+    if (!auth.isSignedIn.get()) {
       return false;
     }
-    await authInstance.signOut();
-    localStorage.removeItem('token');
+    await auth.signOut();
     return true;
   }
 
+  private oneTimeBindToSignIn<T>(arg: T, func: (a: T) => void) {
+    // Infinite listeners are allowed, this makes sure each listener runs once.
+    let called = false;
```

Comment:
I'm confused about what this actually does. The variable doesn't seem to be used anywhere

Reply:
It looks like this is being used to ensure that func(args) is only called once. Is that correct?

Reply:
Yes, each listener is a closure and has a reference to their own `called` variable that prevents duplicate callings which could cause errors.